### PR TITLE
updated referral test

### PR DESCRIPTION
**Pull request by ryannguyen16**
# Description

changed the referral test all method to account for new referral_customers keys.

# Testing

updated cassettes. 

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by Justintime50 — APPROVED** [review]
Nice!

**Review by jchen293 — APPROVED** [review]

File: tests/test_referral.py [hunk 1]
```diff
@@ -40,8 +40,11 @@ def test_referral_user_all(prod_api_key, page_size):
     """This test requires a partner user's production API key via EASYPOST_PROD_API_KEY."""
     referral_users = easypost.beta.Referral.all(page_size=page_size)
 
-    assert len(referral_users) <= page_size
-    assert all(isinstance(referral_user, easypost.User) for referral_user in referral_users)
+    referral_users_array = referral_users["referral_customers"]
```
Comment (nwithan8) [inline tests/test_referral.py:43]: Why is this a JSON key rather than a property of an object?

**Review by nwithan8 — CHANGES_REQUESTED** [review]

File: tests/test_referral.py (hunk 1 above)
Comment (Justintime50) [reply tests/test_referral.py:43]: Because all of our tests are written this way?

Comment (nwithan8) [reply tests/test_referral.py:43]: I see, we don't have explicit "list" or "collection" wrapper objects, so we have to [look up a key in a dictionary](https://github.com/EasyPost/easypost-python/blob/5ede5f7a1ea4fa6ad9c20418eb190010ca89caf0/tests/test_shipment.py#L32). Ugh, this library.

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**